torch/nn/quantized/modules/linear.py

```diff
@@ -150,13 +151,18 @@ def from_float(mod):
             Args: `mod` a float module, either produced by torch.quantization utilities
             or directly from user
         """
-        assert type(mod) == NNLinear, 'nnq.Linear.from_float only works for nn.Linear'
-        assert hasattr(mod, 'qconfig'), 'Input float module must have qconfig defined'
-        assert hasattr(mod, 'observer'), 'Input float module must have observer attached'
-        activation_observer = mod.observer
+        if mod.training:
```

Comment:
it's a bit weird to rely here on training/eval, isn't it? maybe just check what it was instrumented with? (i.e. whether it's weight_fake_quant or not

Reply:
why? we are already using this to control the overall flow

Reply:
@jerryzh168 : What if we just wanted to insert fake-quants for weights to simulate accuracy for eval? Will this be too restrictive in that case?

Reply:
not for this place in particular, this is called in last convert step, if we just insert fake-quant we won't hit this code.